packages/ckb-sdk-core/src/signWitnessGroup.ts

```diff
@@ -39,10 +57,18 @@ const signWitnessGroup = (
   if (typeof sk === 'string') {
     const keyPair = new ECPair(sk)
     emptyWitness.lock = keyPair.signRecoverable(message)
+    return [multisigConfig ? emptyWitness : serializeWitnessArgs(emptyWitness), ...witnessGroup.slice(1)]
   } else {
-    emptyWitness.lock = sk(message)
+    const skResult = sk(message, [emptyWitness, ...witnessGroup.slice(1)])
+    if (typeof skResult === 'string') {
+      emptyWitness.lock = skResult
+      return [multisigConfig ? emptyWitness : serializeWitnessArgs(emptyWitness), ...witnessGroup.slice(1)]
+    }
+    return skResult.then(res => {
+      emptyWitness.lock = res
+      return [multisigConfig ? emptyWitness : serializeWitnessArgs(emptyWitness), ...witnessGroup.slice(1)]
+    })
```

Comment:
Not covered by tests
Ref: https://app.codecov.io/gh/nervosnetwork/ckb-sdk-js/compare/593/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr%20comments&utm_term=nervosnetwork#D3L67